src/lib/netlist/plib/ptests.h

```diff
@@ -51,7 +51,7 @@
 
 #define PINT_EXPECT(comp, exp1, exp2) \
 	if (!plib::testing::internal_assert(plib::testing::comp_ ## comp (), # exp1, # exp2, exp1, exp2)) \
-		std::cout << __FILE__ << ":" << __LINE__ << ":1: error: test failed\n"
+		std::cout << std::string(__FILE__) << std::string(":") << __LINE__ << std::string(":1: error: test failed\n")
```

Comment:
Is there a good reason for constructing `std::string` here?  It potentially causes allocations.

Reply:
I had issues with getting overloaded `<<` stream operators to compile with apple clang. This is a left-over. I will remove std::string and test the compile again. Afterwards I will commit.

> It’s good if some of the issues in PPMF can be resolved.
> 
>     * Have you fully exercised this for a target using the ARM variant of the Itanium ABI (e.g. ARM, AArch64, or WebAssembly)?

No. It would be great if somebody could point me to an image I can e.g. use with QEMU to compile netlist. In the mean time it would be great if somebody with access to such a system could do.
```
cd src/lib/netlist/build
make
./nltool --cmd tests

>     * At a glance, I didn’t notice provision for targets where function pointers point to descriptors (e.g. AIX/WinNT PPC, Linux/AIX PPC64, Itanium, etc.).  Are you catering for this?

Please see above. 
 
>     * Did you consider walking resolved PLT entries and out-of-line virtual call stubs used with MSVC ABI to reduce overhead?

Yes. The code is currently working and I will consider this when mame will compile under MSVC again. For some reason (compiler bug) this is currently not the case.
 
>     * I take it you’ve read [this](https://rants.vastheman.com/2021/09/21/msvc/) which explains the MSVC ABI for pointers to member functions in some detail.

Will do :-) prior to implementing resolved PLT.

Reply:
How small/isolated can you make the PPMF tests themselves?  A Raspberry Pi running Linux can be used for testing small stuff against the ARM variant of the Itanium ABI.

Reply:
The standalone netlist build is pretty small. I anyhow need an replacement PI 4 for my homeautomation. The PI 3 is now a couple of years old. 
But ... there aren't any PIs available at reasonable prices. Once I get one I first use it to test the code.

Reply:
> The standalone netlist build is pretty small. I anyhow need an replacement PI 4 for my homeautomation. The PI 3 is now a couple of years old.

A Pi3 should be fine just for testing PPMF though.  (I’d argue a Pi3 is a better choice than a Pi4 for a lot of applications, just because of the lower power consumption and heat output.)

> But ... there aren't any PIs available at reasonable prices. Once I get one I first use it to test the code.

Yeah, scalpers ruin everything.  I expect at least a few other MAME developers do have Pi models with build environments set up that they could build a minimal test on, though.